### ISSUE TITLE
SonarJS no longer reporting code coverage for Typescript

### ISSUE THREAD

**Issue by tanepiper** [SonarSource/SonarJS#1972]
**SonarJS version:** 5.2.1 (build 7778)<br>
**SonarQube version:** Community Version 7.9.2 (build 30863) & Version 7.0 (build 36138)<br>

Between March 6th and Today, our pipeline is no longer reporting code coverage - either in full or on new code.

I'm also testing this locally using a local docker instance and `sonarqube-scanner` npm module @ `2.5.0`

We have a multi-module setup for an Angular project that has a root properties file and uses the multi-module for each library and app shipped from this repo

Nothing has changed in our configuration files, but we are now observing the following message in the sonar output:

```
INFO: Analysing [/Users/tpiper1/work/frontend/core-frontend/libs/core/auth-core/../../../coverage/libs/core/auth-core/lcov.info]
WARN: Could not resolve 7 file paths in [/Users/tpiper1/work/frontend/core-frontend/libs/core/auth-core/../../../coverage/libs/core/auth-core/lcov.info], first unresolved path: libs/core/auth-core/src/lib/auth-core.module.ts
```

**Reproducer** 
The main sonar file is:
```
sonar.projectKey=core:frontend
sonar.projectName=Core Frontend
sonar.exclusions=**/node_modules/**/*, **/*.spec.ts, **/*.e2e.ts, **/environments/**/*, **/*.routes.ts, **/main.ts, **/test-setup.ts, **/polyfills.ts
sonar.test.inclusions=**/*.spec.ts
sonar.sourceEncoding=UTF-8
sonar.typescript.tsconfigPath=tsconfig.json
sonar.host.url=http://localhost:9000 # Normally the remote URL
sonar.modules=api-core,auth-core,utils-core,demo-app
api-core.sonar.projectBaseDir=libs/core/api-core
auth-core.sonar.projectBaseDir=libs/core/auth-core
utils-core.sonar.projectBaseDir=libs/core/utils-core
demo-app.sonar.projectBaseDir=apps/demo/demo-app
```

An example submodule would be:

```
sonar.projectName=Utilities Core
sonar.sources=src
sonar.tests=src
sonar.typescript.lcov.reportPaths=../../../coverage/libs/core/utils-core/lcov.info
```

If I change to `sonar.typescript.lcov.reportPaths` to `sonar.javascript.lcov.reportPaths` I don’t get the above issue, but there is still zero coverage.

**Logs**
```
INFO: Sensor SonarTS [typescript]
INFO: Analyzing 11 typescript file(s) with the following configuration file 
/Users/tpiper1/work/frontend/core-frontend/libs/core/auth-core/tsconfig.json
INFO: 11 files analyzed out of 11
INFO: Sensor SonarTS [typescript] (done) | time=3634ms
```

Before:
![Screenshot 2020-03-18 at 18 34 56](https://user-images.githubusercontent.com/376930/76991342-b9dcb600-6949-11ea-841b-dc7d1024e7ae.png)

After:
![Screenshot 2020-03-18 at 18 35 33](https://user-images.githubusercontent.com/376930/76991361-c2cd8780-6949-11ea-8a96-048c70ad458b.png)

**Comment by tanepiper** [comment]
In an attempt to isolate the issue, I updated to Sonarqube 8.2 locally and it's reverted back to `Unresolved paths`.  I've attached the verbose log:

[output.log.txt](https://github.com/SonarSource/SonarJS/files/4351770/output.log.txt)

**Comment by tanepiper** [comment]
OK!  So I've been able to confirm with the following settings:

SonarQube 8.20.32929 Community Edition
SonarQube Scanner 4.0.0.1744 (installed via `sonarqube-scanner` npm module)
Java 11.0.3 AdoptOpenJDK (64-bit)
Mac OS X 10.14.6 x86_64

If I manually go to the lcov.info file and make the `SF:` paths absolute then it reports coverage! It seems to be a problem with relative paths, which suggests a regression of 49f34ea (Release 5.2.1)

![Screenshot 2020-03-19 at 01 22 19](https://user-images.githubusercontent.com/376930/77019176-3db19500-6980-11ea-998a-ac364f09ccf3.png)
![Screenshot 2020-03-19 at 01 22 43](https://user-images.githubusercontent.com/376930/77019180-3f7b5880-6980-11ea-86fd-c18066cae498.png)

**Comment by tanepiper** [comment]
Mentioning #1264 as this seems to be a further regression.

I'm not a Java dev, but I had a little dig in to the code and https://github.com/SonarSource/SonarJS/pull/1263/files#diff-3467835f27e792af2b2accfa678686f4R155 is where the issues seems to lie.

`context.fileSystem().predicates().hasPath(filePath)` returns a RelativePathPredicate, but `context.fileSystem().inputFile` returns null from that value

When calling `fileLocator.getInputFile` the return value:

`return tree.getFileWithSuffix(pathElements);`

is again returning null on the path elements.

After a bit more digging the culprit seems to be `ReversePathTree.getFileWithSuffix`

For a path `libs/core/auth-core/src/lib/store/auth.store.ts` if I output the index and `currentNode.children.get(path[i])` value it looks like this for all 

```
6 auth.store.ts 
5 store
4 lib 
3 src 
2 auth-core
null
```

I renamed the path to `authcore` and it seems that this method never then gets called, so I'm assuming earlier in the Lcov parsing something doesn't return null.  I've included another output log (as it's still not actually generating any code coverage values) but it appears #1264 made the issue for relative paths worse :(

[out.log.txt](https://github.com/SonarSource/SonarJS/files/4354395/out.log.txt)
